underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -186,12 +197,13 @@ public boolean delete(String path, boolean recursive) throws IOException {
 
   @Override
   public boolean exists(String path) throws IOException {
+    //TODO(adit): this hack should not be here
     // To get better performance Swift driver does not create a _temporary folder.
     // This optimization should be hidden from Spark, therefore exists _temporary will return true.
     if (path.endsWith("_temporary")) {
       return true;
     }
-    return getObject(path).exists();
+    return isRoot(path) || isFile(path) || isDirectory(path);
```

Comment:
this is inefficient for a non-existent object we end up invoking `isRoot` and more importantly `getObject` twice ... this could be made more efficient by stripping the folder suffix from `path` (which will be a no-op for files) and invoking `getObject` directly